### PR TITLE
Fix divider placement on reversed axes

### LINKED ISSUE #4927: Multicategory axis dividers are misaligned when autorange is reversed
This happens on any axis where `type = 'multicategory'` and `autorange = 'reversed'`.

### PR DESCRIPTION
**Pull request by justin-sleep**
Fixes #4927. The divider placement algorithm now uses the correct boundary when the axis is reversed.

### CODE REVIEW

**Comment by archmoj** [comment]
@justin-sleep thanks very much for the PR.
Good catch!
Could you please add a mock for image test here: https://github.com/plotly/plotly.js/tree/master/test/image/mocks?

File: src/plots/cartesian/axes.js [hunk 1]
```diff
@@ -2100,6 +2100,9 @@ function getDividerVals(ax, vals) {
     var out = [];
     var i, current;
 
+    var xvals = vals.map(a => a.x);
```
Comment (archmoj) [inline src/plots/cartesian/axes.js:2103]: Could you please rewrite this line in ES5?

**Comment by justin-sleep** [comment]
Sure, those changes have been made.

**Comment by archmoj** [comment]
> Sure, those changes have been made.

Awesome. Let's see the test results...

**Comment by archmoj** [comment]
Would you mind replacing the new baseline with the one below which is created in a docker?
reversed-axis-dividers
Thank you!

**Comment by archmoj** [comment]
We need to figure out why few tests are failing.

**Comment by archmoj** [comment]
:tada: Great! LGTM.
Now that the CI as well as `no_ci` tests passed (on my machne), let me request a review from @alexcjohnson.

**Comment by archmoj** [comment]
> This fix in `getDividerVals` looks great. There's similar code in `getBoundaryVals`, does it need similar reversed-range handling or does it already work correctly? I've lost track of the distinction between these functions.

It seems to me that a similar logic may be needed in `getBoundaryVals`.

**Pull request merged**